### PR TITLE
Feature/new config method

### PR DESCRIPTION
**Pull request by reaganthomas**

### CODE REVIEW

File: templates/server/config/environment/doNotCommit.env.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+(function IIFE() {
+  'use strict';
+
+  // Environment variables and production values not to be comitted to repo but needed for development.
+  const config = {
+    // CLIENT_SECRET: 345912384fjwasif3p9w8fj (aws secret)
```
Comment (joshua-gesto) [inline templates/server/config/environment/doNotCommit.env.js]: So, these are not actually being added to `process.env`. Which means that all modules need to require the config file. It's fine, but some services (like [the auth service](https://github.com/Gesto/gesto-auth-middleware/blob/develop/lib/database/index.js#L10)) still look for env vars.

File: templates/server/config/environment/test.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
   module.exports = {
     <% if(mongoose) { %>mongo: {
-      uri: 'mongodb://localhost/<%= appNameSlug %>-test'
+      uri: 'mongodb://localhost/<%= appNameSlug %>'
```
Comment (joshua-gesto) [inline templates/server/config/environment/test.js]: Won't this mess up the practice of using different DBs for test vs. dev?

Comment (reaganthomas) [inline templates/server/config/environment/test.js]: It's more of something you opt into on purpose I would think. This is just a generator, so you should still set it up the way that works for that project. Right now we don't have anything that necessitates a test db sep from develop, but if a project did it's an easy change.

Comment (prestonvanloon) [inline templates/server/config/environment/test.js]: I think this is helpful to have test run on a different DB. Can we change it back?

**Comment by bsidelinger912** [comment]
One thing this doesn't cover is how to set process.env.JWT_SECRET (soon to become process.env. JWT_SIGNING_KEY) to support the new auth middleware.  That is an environment var that has to be set explicitly as auth middleware doesn't look into your app's config.  It'd be nice to show some sort of example for how one would do that, I'm think we'd just set it in the doNotCommit.env.js file.

**Comment by jpringle** [comment]
@bsidelinger912 interesting point - should it be ok for modules to require specific environment variables, or should they expose required configuration objects that would be populated with such?  Ie, in this example should the auth-middleware just expose some configuration attribute for signing key vs a specific environment variable?

The benefit I see of having the module define/declare the environment variables upon which it depends is that the definition is centralized.

The downside I see is that it's non-obvious that the application now requires an additional set of environment variables, simply because a particular module is included.  One would expect to understand all required/available environment variables by looking at the environment file(s).

File: templates/server/config/environment/test.js (hunk 2 above)
Comment (reaganthomas) [inline templates/server/config/environment/test.js]: ok

**Pull request merged**